### PR TITLE
tickets/SITCOM-912

### PR DESCRIPTION
**Pull request by mareuter**
Add git index corruption troubleshooting for argocd.

### CODE REVIEW

File: docs/applications/argo-cd/troubleshoot.rst [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+If an ArgoCD app or apps are giving the following error in the UI
+
+.. code-block:: shell
+
+    rpc error: code = Internal desc = Failed to fetch default: `git fetch origin --tags --force --prune` failed exit status 128: fatal: .git/index: index file smaller than expected
+
+The git index for the cloned repository that controls the apps is corrupted and needs to be removed.
+Do this by the following:
+
+#. Find the argocd-repo-server pod and grep the logs.
```
Comment (jonathansick) [inline docs/applications/argo-cd/troubleshoot.rst:21]:
```suggestion
#. Find the ``argocd-repo-server`` pod and grep the logs:
```

**Review by jonathansick — APPROVED** [review]
Thanks for this. I just have some minor edits for our style and one possible point to clarify regarding parsing the grep output.

File: docs/applications/argo-cd/troubleshoot.rst [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+    .. code-block:: shell
+
+        pod=$(kubectl get pods -n argocd -l app.kubernetes.io/name=argocd-repo-server | grep argocd | awk '{print $1}')
+        kubectl logs -n argocd $pod | grep -B1 "index file smaller than expected" | grep -B1 "execID"
+
+#. Find the "dir" entry in the line above the one has the phase noted in the previous step. Example output show below.
```
[the remark this answers is not in the document]
Comment (mareuter) [reply docs/applications/argo-cd/troubleshoot.rst:28]: Should have been "phrase".

[the remark this answers is not in the document]
Comment (mareuter) [reply docs/applications/argo-cd/troubleshoot.rst:28]: I think your statement is clear and succinct, so I'll use that.

File: docs/applications/argo-cd/troubleshoot.rst [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+==========================
+
+If an ArgoCD app or apps are giving the following error in the UI
+
+.. code-block:: shell
+
+    rpc error: code = Internal desc = Failed to fetch default: `git fetch origin --tags --force --prune` failed exit status 128: fatal: .git/index: index file smaller than expected
+
+The git index for the cloned repository that controls the apps is corrupted and needs to be removed.
+Do this by the following:
```
[the remark this answers is not in the document]
Comment (mareuter) [reply docs/applications/argo-cd/troubleshoot.rst]: Fixed.

File: docs/applications/argo-cd/troubleshoot.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (mareuter) [reply docs/applications/argo-cd/troubleshoot.rst]: Fixed.

File: docs/applications/argo-cd/troubleshoot.rst [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+        kubectl logs -n argocd $pod | grep -B1 "index file smaller than expected" | grep -B1 "execID"
+
+#. Find the "dir" entry in the line above the one has the phase noted in the previous step. Example output show below.
+
+    .. code-block:: shell
+
+        {"dir":"/tmp/_argocd-repo/35fe76f8-488a-4871-baaa-5f81d81331b1","execID":"a98af","level":"info","msg":"git fetch origin --tags --force --prune","time":"2023-06-13T18:48:12Z"}
+        {"execID":"a98af","level":"error","msg":"`git fetch origin --tags --force --prune` failed exit status 128: fatal: .git/index: index file smaller than expected","time":"2023-06-13T18:48:12Z"}
+
+#. Exec into the repo server pod
```
[the remark this answers is not in the document]
Comment (mareuter) [reply docs/applications/argo-cd/troubleshoot.rst]: Fixed.

File: docs/applications/argo-cd/troubleshoot.rst (hunk 1 above)
Comment (mareuter) [reply docs/applications/argo-cd/troubleshoot.rst]: Fixed.

**Pull request merged**